Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -706,4 +706,23 @@ def test_irb
 
     assert_match "Interactive Homebrew Shell", cmd("irb", irb_test)
   end
+
+  def test_analytics
+    assert_match "Invalid usage", cmd_fail("analytics", "on", "off")
+    assert_match "Invalid usage", cmd_fail("analytics", "testball")
+    assert_match "Analytics is enabled", cmd("analytics")
+    assert_match "Analytics is disabled",
+      cmd("analytics", "HOMEBREW_NO_ANALYTICS" => "1")
+
+    HOMEBREW_REPOSITORY.cd do
+      shutup do
+        system "git", "init"
+      end
+    end
```

Comment:
Thanks for the detailed response! I'm happy with the conclusion you drew from this (moving the repository initialization to the top of the test). :+1: